modules/sdk/src/main/scala/org/tessellation/sdk/domain/Daemon.scala

```diff
@@ -14,4 +20,18 @@ object Daemon {
     def start: F[Unit] = S.supervise(thunk).void
   }
 
+  def periodic[F[_]: Async](thunk: F[Unit], sleepTime: FiniteDuration)(implicit S: Supervisor[F], Logger: Logger[F]): Daemon[F] =
+    new Daemon[F] {
+      def start: F[Unit] =
+        S.supervise(
+          Stream
+            .awakeEvery(sleepTime)
+            .evalMap { _ =>
+              thunk.handleErrorWith(err => Logger.error(err)("The daemon encountered an error."))
```

Comment:
what about passing the logger explicitly and naming it `errorLogger`? 🤔

Reply:
That code was removed.